src/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -260,6 +266,43 @@ public sealed partial class JsonNumber : System.Text.Json.JsonNode, System.IEqua
         [System.CLSCompliantAttribute(false)]
         public bool TryGetUInt64(out ulong value) { throw null; }
     }
+    public sealed partial class JsonObject : System.Text.Json.JsonNode, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>>, System.Collections.IEnumerable
+    {
+        public JsonObject(System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, System.Text.Json.JsonNode>> jsonProperties, System.Text.Json.DuplicatePropertyNameHandling duplicatePropertyNameHandling = System.Text.Json.DuplicatePropertyNameHandling.Replace) { }
+        public JsonObject(System.Text.Json.DuplicatePropertyNameHandling duplicatePropertyNameHandling = System.Text.Json.DuplicatePropertyNameHandling.Replace) { }
```

Comment:
I wouldn't have expected `DuplicatePropertyNameHandling` to show up within JsonObject's API surface, but rather within the options passed in to the type that creates JsonObjects from some json payload (say `JsonDocumentOptions`).

Do we think there will be a need for this setting when creating a JsonObject itself via the ctor? What scenario would it be used for?

Reply:
So actually, my `Add` behaves as specified in `DuplicatePropertyNameHandling`, to be consistent. So while indexer will always replace values, `Add` will either throw / replace or ignore (defaultly would behave just as indexer, but I can change the default behavior to exceptions). 

Maybe we could have those options both in `JsonObject` and passed to parsing method? I think people might want to specify the behavior they want also when creating objects from C# - what this API is designed for?

Reply:
> I think people might want to specify the behavior they want also when creating objects from C# - what this API is designed for?

Can you give an example/sample where you think it would be useful to have `DuplicatePropertyNameHandling` when creating `JsonObject`?

Reply:
For example something containing interactions with user?

```csharp
var properties = new JsonObject(DuplicatePropertyNameHandling.Error);

while(PropertiesReady())
{
    string userProvidedPropertyName = Console.ReadLine();
    string userProvidedPropertyValue = Console.ReadLine();
    try
    {
        properties.Add(userProvidedPropertyName, userProvidedPropertyValue);
    }
    catch (ArgumentException)
    {
        Console.WriteLine(string.Format("Property name `{0}` already exists. Please provide properties with unique names.", userProvidedPropertyName));
    }
}
Mailbox.SendData(properties.AsJsonElement());
```